src/components/TodoItem.tsx

```diff
@@ -0,0 +1,102 @@
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+import classNames from 'classnames';
+import { Todos } from '../types/Todos';
+import { TodoContext } from '../contexts/TodoContext';
+
+type Props = {
```

Comment:
Use interface prefer.
Type has unique ability but if you don`t need them use interface. It is sampled in scaled and has more interesting features.